src/sentry/api/endpoints/custom_rules.py

```diff
@@ -151,3 +141,78 @@ def post(self, request: Request, organization: Organization) -> Response:
             )
         except ValueError as e:
             return Response({"query": ["Could not convert to rule", str(e)]}, status=400)
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        requested_projects = request.GET.getlist("project")
+        query = request.GET.get("query")
+
+        try:
+            requested_projects_ids = [int(project_id) for project_id in requested_projects]
+        except ValueError:
+            return Response({"projects": ["Invalid project id"]}, status=400)
+
+        if requested_projects_ids:
+            org_rule = False
+            invalid_projects = []
+            available_projects = {
+                p.id for p in Project.objects.get_many_from_cache(requested_projects_ids)
+            }
+            for project_id in requested_projects_ids:
+                if project_id not in available_projects:
+                    invalid_projects.append(f"invalid project id: {project_id}")
+
+            if invalid_projects:
+                raise serializers.ValidationError({"projects": invalid_projects})
+        else:
+            # no project specified (it is an org rule)
+            org_rule = True
+
+        try:
+            tokens = parse_search_query(query)
+        except InvalidSearchQuery as e:
+            return Response({"query": [str(e)]}, status=400)
```

Comment:
I would prefer to expose a generic error message instead of just dumping the actual error.

Reply:
I think having an error that actually tells you what's wrong is preferable.
This is something related to the user input so it makes sense to emit this.
Ideally I can distill this into a user message ( not a generic error) but at the moment I don't have enough info
about it.

Reply:
It's mainly a security concern but if you know that the message will be always bounded and won't expose any internals this is fine.